### PR TITLE
 Updated original content of debug-wordpress.md

### PR DESCRIPTION
**Pull request by cpaloukas**
reviewed current content
added a note for using a staging environment or backing up the site prior to changes
added Custom PHP Debugging section
updated Debugging Plugins section with descriptions for each plugin

### CODE REVIEW

File: debug/debug-network.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+If you have reached this page, chances are you have received an error in your [WordPress network](https://wordpress.org/documentation/article/multisite-network-administration/). This failure occurs when WordPress cannot find one or more of the global tables for the network in the [database](https://codex.wordpress.org/Database_Description).
 
-On some shared webhosts, the host has disabled the functionality from running. It is always best to check with your webhost **before** [creating a network](https://developer.wordpress.org/advanced-administration/wordpress/wp-config/) to make sure your webhost account fulfills the technical requirements.
+On some shared webhosts, the host has disabled the functionality from running. It is always best to check with your webhost **before** [creating a network](https://github.com/WordPress/Advanced-administration-handbook/blob/main/multisite/create-network.md) to make sure your webhost account fulfills the technical requirements.
```
Comment (Crixu) [inline debug/debug-network.md:7]: The handbook will be transferred over to wp.org and the links should still point there.

Comment (javiercasares) [reply debug/debug-network.md:7]: The future link for “prepare a network” is:
https://developer.wordpress.org/advanced-administration/multisite/prepare-network/

The future link for “create a network” is:
https://developer.wordpress.org/advanced-administration/multisite/create-network/

File: debug/debug-network.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
-## If You just installed your network {#if-you-just-installed-your-network}
+## If You just installed your network
 
-Check your [wp-config.php](https://developer.wordpress.org/advanced-administration/wordpress/wp-config/) file for :
+Check your [wp-config.php](https://github.com/WordPress/Advanced-administration-handbook/blob/main/wordpress/wp-config.md) file for :
```
Comment (javiercasares) [inline debug/debug-network.md:11]: The future link for “wp-config” is:
https://developer.wordpress.org/advanced-administration/wordpress/wp-config/

File: debug/debug-network.md [hunk 3]
```diff
@@ -105,14 +105,11 @@ This will result in a Network install appearing to fail and may show WP errors l
...
 ```
 
-## Related Articles {#related-articles}
+## Related Articles
 
-* [Create A Network](https://developer.wordpress.org/advanced-administration/multisite/create-network/)
-
-## External Links {#external-links}
-
-* [WordPress › Support » Multisite](https://wordpress.org/documentation/forum/multisite)
```
Comment (javiercasares) [inline debug/debug-network.md:114]: The URL for the forum is:
https://wordpress.org/support/forum/multisite/

File: debug/debug-wordpress.md [hunk 4]
```diff
@@ -4,6 +4,8 @@ Debugging PHP code is part of any project, but WordPress comes with specific deb
 
 For non-programmers or general users, these options can be used to show detailed information about errors.
 
+**NOTE**: Prior to making any modifications to your website, verify that you have either utilized a staging environment or taken an appropriate backup of your site.
+
 ## Example wp-config.php for Debugging
 
 The following code, inserted in your [wp-config.php](https://wordpress.org/documentation/article/editing-wp-config-php/) file, will log all errors, notices, and warnings to a file called `debug.log` in the wp-content directory. It will also hide the errors so they do not interrupt page generation.
```
Comment (javiercasares) [inline debug/debug-wordpress.md:11]: The new URL for “wp-config” is:
https://developer.wordpress.org/advanced-administration/wordpress/wp-config/

File: debug/debug-wordpress.md [hunk 5]
```diff
@@ -55,6 +57,32 @@ Enabling `WP_DEBUG` will cause all PHP errors, notices and warnings to be displa
 
 Showing all PHP notices and warnings often results in error messages for things that don’t seem broken, but do not follow proper data validation conventions inside PHP. These warnings are easy to fix once the relevant code has been identified, and the resulting code is almost always more bug-resistant and easier to maintain.
 
+### Custom PHP Debugging
+
+If it is necessary to log non-error information for debugging purposes, PHP does offer the `error_log` function for this purpose. However, this method does not provide properly formatted output by default.
+
+To address this you may add an additional function on your site to handle formatting either by creating a [custom plugin](https://developer.wordpress.org/plugins/plugin-basics/) or using a snippet by leveraging the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin. The function will act as a wrapper for the `error_log` using `print_r`  to format arrays and objects correctly before logging them.
```
Comment (javiercasares) [inline debug/debug-wordpress.md:64]: We shouldn't recommend plugins.

File: debug/debug-wordpress.md [hunk 6]
```diff
@@ -114,14 +142,19 @@ The array is stored in the global `$wpdb->queries`.
 There are many [debugging plugins](https://wordpress.org/plugins/search.php?q=debug) for WordPress that show more information about the internals, either for a specific component or in general. Here are some examples:
 
 * [Query Monitor](https://wordpress.org/plugins/query-monitor/)
+    * Query Monitor is described as ‘the developer tools panel for WordPress’. It enables debugging of database queries, PHP errors, hooks and actions, block editor blocks, enqueued scripts and stylesheets, HTTP API calls, and more.
+    As well as the above it includes some advanced features such as debugging of Ajax calls, REST API calls, and user capability checks. It provides the ability to narrow down much of its output by plugin or theme, allowing you to quickly determine poorly performing plugins, themes, or functions.
 * [Debug Bar](https://wordpress.org/plugins/debug-bar/)
+    * Debug Bar adds a debug menu to the admin bar that shows query, cache, and other helpful debugging information. When WP_DEBUG is enabled it also tracks PHP Warnings and Notices to make them easier to find.
 * [Log Deprecated Notices](https://wordpress.org/plugins/log-deprecated-notices/)
```
Comment (javiercasares) [inline debug/debug-wordpress.md:149]: We shouldn't recommend deprecated plugins.

File: debug/debug-wordpress.md [hunk 7]
```diff
@@ -114,14 +142,19 @@ The array is stored in the global `$wpdb->queries`.
...
+    * Log Deprecated Notices tool logs the usage of deprecated files, functions, and arguments, helping to pinpoint where deprecated elements are being used and presenting a replacement option if it exists. 
 
 ## External Resources
 
 * [WordPress 'wp-config.php' file Generator](http://generatewp.com/wp-config/)
 * ['No White Screen' plugin: Display the error instead of a white screen](https://github.com/stracker-phil/wp-no-white-screen/)
```
Comment (javiercasares) [inline debug/debug-wordpress.md:155]: This look like deprecated.

File: debug/debug-wordpress.md [hunk 8]
```diff
@@ -114,14 +142,19 @@ The array is stored in the global `$wpdb->queries`.
...
+    As well as the above it includes some advanced features such as debugging of Ajax calls, REST API calls, and user capability checks. It provides the ability to narrow down much of its output by plugin or theme, allowing you to quickly determine poorly performing plugins, themes, or functions.
 * [Debug Bar](https://wordpress.org/plugins/debug-bar/)
+    * Debug Bar adds a debug menu to the admin bar that shows query, cache, and other helpful debugging information. When WP_DEBUG is enabled it also tracks PHP Warnings and Notices to make them easier to find.
 * [Log Deprecated Notices](https://wordpress.org/plugins/log-deprecated-notices/)
+    * Log Deprecated Notices tool logs the usage of deprecated files, functions, and arguments, helping to pinpoint where deprecated elements are being used and presenting a replacement option if it exists. 
 
 ## External Resources
 
 * [WordPress 'wp-config.php' file Generator](http://generatewp.com/wp-config/)
```
Comment (javiercasares) [inline debug/debug-wordpress.md:154]: Maybe an HTTP version:
https://generatewp.com/wp-config/

Comment (javiercasares) [reply debug/debug-wordpress.md:154]: We can see if there are other tools like:
https://wpconfig.wpsysadmin.com/

File: debug/index.md [hunk 9]
```diff
@@ -1,9 +1,31 @@
 # Debugging WordPress
 
+Debugging is an essential part of website development and a crucial step in ensuring a smooth and reliable experience for users. Debugging refers to the process of finding and fixing errors or issues within a website's code. 
 
+In this part of the Avdanced Administration Handbook, we will address various aspects of debugging a WordPress website, including general WordPress debugging, JavaScript debugging, WordPress network debugging, and test-driving.
 
+## General WordPress Debugging
 
+When it comes to debugging a WordPress site, there are various methods and techniques to use including turning on debugging in the WordPress configuration file, using error logs and the use of debugging plugins. These techniques can help identify and resolve various types of errors such as PHP errors and [database](https://codex.wordpress.org/Database_Description) errors.
```
Comment (javiercasares) [inline debug/index.md:9]: “Debug WordPress” will be at:
https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/

File: debug/index.md [hunk 10]
```diff
@@ -1,9 +1,31 @@
...
+
+## Javascript Debugging
+
+[JavaScript](https://codex.wordpress.org/Using_Javascript) debugging is essential for ensuring that the site's front-end interactions and user experience are functioning correctly. Using browser dev tools to inspect the source code and diagnose issues with JavaScript along with a few more tips are provided in this setion.
```
Comment (javiercasares) [inline debug/index.md:13]: "Debug JavaScript" will be at:
https://developer.wordpress.org/advanced-administration/debug/debug-javascript/

File: debug/index.md [hunk 11]
```diff
@@ -1,9 +1,31 @@
...
+
+## WordPress Network Debugging
+
+[WordPress network](https://wordpress.org/documentation/article/multisite-network-administration/) debugging is a required when maintaining a network of WordPress sites; referring to the process of debugging multiple sites that are connected through a single installation of WordPress. This type of setup is often used by organizations that have multiple websites that share a common set of plugins, themes, and users.
```
Comment (javiercasares) [inline debug/index.md:17]: “Debug Network” will bet at:
https://developer.wordpress.org/advanced-administration/debug/debug-network/

File: debug/index.md [hunk 12]
```diff
@@ -1,9 +1,31 @@
...
+## Javascript Debugging
+
+[JavaScript](https://codex.wordpress.org/Using_Javascript) debugging is essential for ensuring that the site's front-end interactions and user experience are functioning correctly. Using browser dev tools to inspect the source code and diagnose issues with JavaScript along with a few more tips are provided in this setion.
+
+## WordPress Network Debugging
+
+[WordPress network](https://wordpress.org/documentation/article/multisite-network-administration/) debugging is a required when maintaining a network of WordPress sites; referring to the process of debugging multiple sites that are connected through a single installation of WordPress. This type of setup is often used by organizations that have multiple websites that share a common set of plugins, themes, and users. 
+
+## Test-driving
+Test-driving refers to the process of testing a website before making it live. This is  process allows developers to identify and resolve any issues or bugs before the site is made available to the public. Test-driving is typically performed in a sandbox environment. Creating sandbox environment is covered in this section.
```
Comment (javiercasares) [inline debug/index.md:20]: “Test driving” will be at:
https://developer.wordpress.org/advanced-administration/debug/test-driving/

File: debug/index.md [hunk 13]
```diff
@@ -1,9 +1,31 @@
...
+
+## Internal Links
```
Comment (javiercasares) [inline debug/index.md:22]: The internal links probably should be at the sections before...

**Review by javiercasares — CHANGES_REQUESTED** [review]
Some changes pending, mainly URLs

**Review by javiercasares — APPROVED** [review]

File: debug/debug-network.md [hunk 14]
```diff
@@ -105,14 +105,15 @@ This will result in a Network install appearing to fail and may show WP errors l
 One or more database tables are unavailable. The database may need to be repaired.
 ```
 
-## Related Articles {#related-articles}
+## Related Articles
 
-* [Create A Network](https://developer.wordpress.org/advanced-administration/multisite/create-network/)
+* [WordPress Multisite Network: A Complete Guide](https://multilingualpress.org/wordpress-multisite-network/)
```
Comment (javiercasares) [inline debug/debug-network.md:110]: No external links, pelase.
https://make.wordpress.org/docs/tag/external-linking-policy/

**Review by javiercasares — CHANGES_REQUESTED** [review]
There are some changes needed.

**Pull request merged**